tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZoneSuite.scala

```diff
@@ -0,0 +1,170 @@
+    val s = Instant.parse("%04d-01-01T00:00:00z".format(startYear)).getEpochSecond
+    val e = Instant.parse("%04d-01-01T00:00:00z".format(endYear)).getEpochSecond
+    val epochSeconds = mutable.ArrayBuffer[Long]()
+    for (epoch <- s until e by TimeUnit.MINUTES.toSeconds(15)) {
+      epochSeconds += epoch
+    }
+    // Note: in seconds, not micro seconds
+    epochSeconds.toArray
+  }
+
+  def testFromUTC(epochSeconds: Array[Long], zoneStr: String): Unit = {
+    // get result from Spark
+    val sparkRet = withCpuSparkSession(
+      spark => {
+        createDF(spark, epochSeconds).createOrReplaceTempView("tab")
+        // refer to https://spark.apache.org/docs/latest/api/sql/#from_utc_timestamp
+        // convert from UTC timestamp
+        // first cast long value as timestamp
+        spark.sql(s"select from_utc_timestamp(cast(ts_long as timestamp), '$zoneStr') from tab").collect()
+      },
+      new SparkConf()
+        // use UTC time zone to create date frame
+        .set("spark.sql.session.timeZone", "UTC")
+        // by setting this, the Spark output for datetime type is java.time.Instant instead
+        // of java.sql.Timestamp, it's convenient to compare result via java.time.Instant
+        .set("spark.sql.datetime.java8API.enabled", "true"))
+
+    // get result from TimeZoneDB
+    val actualRet = withResource(createColumnVector(epochSeconds)) { inputCv =>
+      // convert time zone from UTC to specific timezone
+      CpuTimeZoneDB.convertFromUTC(
+        inputCv,
+        ZoneId.of(zoneStr))
+    }
+
+    withResource(actualRet) { _ =>
+      assertRet(actualRet, sparkRet)
+    }
+  }
+
+  def testToUTC(epochSeconds: Array[Long], zoneStr: String): Unit = {
+    // get result from Spark
+    val sparkRet = withCpuSparkSession(
+      spark => {
+        createDF(spark, epochSeconds).createOrReplaceTempView("tab")
+        // refer to https://spark.apache.org/docs/latest/api/sql/#to_utc_timestamp
+        // convert to UTC timezone
+        // first cast long value as timestamp
+        spark.sql(s"select to_utc_timestamp(cast(ts_long as timestamp), '$zoneStr') from tab").collect()
+      },
+      new SparkConf()
+        // use UTC time zone to create date frame
+        .set("spark.sql.session.timeZone", "UTC")
+        // by setting this, the Spark output for datetime type is java.time.Instant instead
+        // of java.sql.Timestamp, it's convenient to compare result via java.time.Instant
+        .set("spark.sql.datetime.java8API.enabled", "true"))
+
+    // get result from TimeZoneDB
+    val actualRet = withResource(createColumnVector(epochSeconds)) { inputCv =>
+      // convert time zone from UTC to specific timezone
+      CpuTimeZoneDB.convertToUTC(
+        inputCv,
+        ZoneId.of(zoneStr))
+    }
+
+    withResource(actualRet) { _ =>
+      assertRet(actualRet, sparkRet)
+    }
+  }
+
+  test("test all time zones") {
+    assume(false,
```

Comment:
Right now we are doing one time zone. Should we have a few time zones enabled by default?  Could we also reduce the number of years that we test so that we don't do the loop 9999 times?  Perhaps every 7 years or every 13 years?

Just as a side note we are going to have the same problem with testing different time zones for all kinds of operators.  We probably want a representative handful op timezones that we try to test in all cases.  We also have the other problem of testing what happens if the default timezone is set differently. This gets to be even harder because we might need to launch a new JVM/python process for each of those to set it properly.

Reply:
> Should we have a few time zones enabled by default?
Added Brazil "America/Sao_Paulo", "Asia/Shanghai".  And randomly select 2 zones.

> Perhaps every 7 years or every 13 years?   

Now it's 7 years.

> We also have the other problem of testing what happens if the default timezone is set differently. 

Our plugin already tested this, the default time zone on executors should be same.